### PR TITLE
Content security policy

### LINKED ISSUE #41: Content Security Policy
#### As a security-minded developer I want to obviate all cross site scripting (XSS) attacks.

A strong content security policy can be used to prevent XSS and other
attacks. With many things security related, it is a good idea to disallow
everything and then allow things only after questioning whether or not they
should be allowed (a whitelist policy).

The most restrictive content security policy is:

    Content-Security-Policy: default-src 'none';

However this does not permit any stylesheets, images, nor javascript sources
all of which we want to be able to serve up locally, but not inline. The
following accomplishes that task:

    Content-Security-Policy: default-src 'none'; script-src 'self'; img-src 'self'; style-src 'self';

Acceptance criteria:
- [x] Every request outputs an appropriate content security policy header.
- [x] There are no content security policy violations in chrome's console.
- [x] The user interface remains the same.

Reference material:
- https://www.owasp.org/index.php/Content_Security_Policy_Cheat_Sheet

Dependencies:
- Image Index

### PR DESCRIPTION
**Pull request by bboe**
Resolves #41.

### CODE REVIEW

File: app/assets/javascripts/views/images/index/index.js [hunk 1]
```diff
@@ -1 +1,9 @@
 window.ImagesIndexView = require('./component').default;
+
+var ready;
+ready = function() {
+  ImagesIndexView.init();
+}
+
+$(document).ready(ready);
+$(document).on('page:load', ready);
```
Comment (rmacklin) [inline app/assets/javascripts/views/images/index/index.js]: I don't think we need either of these actually. We should be able to reduce this file to:
```js
var ImagesIndexView = require('./component').default;
ImagesIndexView.init();
```
(I just did that with Boying today.)

File: config/application.rb [hunk 2]
```diff
@@ -20,6 +20,16 @@ class Application < Rails::Application
     # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
     # config.i18n.default_locale = :de
 
+    csp_settings = ["connect-src 'self'",
+                    "default-src 'none'",
+                    'font-src https://cdn.jsdelivr.net',
+                    "img-src 'self' http: https:",
+                    "script-src 'self' 'unsafe-eval'",
```
Comment (rmacklin) [inline config/application.rb]: Perhaps we should have a comment about the `unsafe-eval` linking to the bootstrap issue.

Comment (bboe) [inline config/application.rb]: Excellent idea.

File: app/assets/javascripts/views/images/index/index.js (hunk 1 above)
Comment (bboe) [inline app/assets/javascripts/views/images/index/index.js]: Cool. Added #44.

**Pull request merged**